daemon/attach.go

```diff
@@ -86,6 +86,7 @@ func (daemon *Daemon) ContainerAttachRaw(prefixOrName string, stdin io.ReadClose
 		CloseStdin: container.Config.StdinOnce,
 	}
 	container.StreamConfig.AttachStreams(&cfg)
+	close(attached)
```

Comment:
It is preventing a potential race where a build could get as far as outputting to std{out,err} before AttachStreams() had actually completed.  In this case, any data outputted by the build would be lost.  I didn't open a separate issue as it's very similar to #31323 (and #29285, which I fixed before); I was hoping just to fix it in passing as it's pretty straightforward.